lib/tasks/retention_schedules.rake

```diff
@@ -1,44 +1,99 @@
+require 'pry'
+
 namespace :retention_schedules do
-  namespace :branston do
-    desc 'create a dummy Offender Sar with retention schedules case'
-    task :create => :environment do
-      if is_on_production?
-        puts "Cannot run this command on production environment!"
-      else
-        puts "Seeding some cases with retention_schedules"
-        states = ['destroy', 'destroy', 'review', 'retain', 'not_set', 'review'] * 3
-
-        count = 0
-        states.each do |state|
-
-          case_with_retention_schedule(
-            case_type: :offender_sar_case,
-            state: state,
-            planned_erasure_date: (Date.today - (3.months + rand(1..10).days))
-          )
-
-          count += 1
-        end
-        puts "Number of cases seeded: #{count}"
+  desc 'create a dummy Offender Sar with retention schedules case'
+  task :create => :environment do
+    if is_on_production?
+      puts "Cannot run this command on production environment!"
+    else
+      puts "Seeding some cases with retention_schedules"
+      states = ['destroy', 'destroy', 'review', 'retain', 'not_set', 'review'] * 3
+
+      count = 0
+      states.each do |state|
+
+        case_with_retention_schedule(
+          case_type: :offender_sar_case,
+          state: state,
+          planned_destruction_date: (Date.today - (3.months + rand(1..10).days))
+        )
+
+        count += 1
       end
+      puts "Number of cases seeded: #{count}"
     end
+  end
+  
+  desc 'Adds retention_schedules to existing closed cases'
+  task :add_rs_to_existing_branston_cases => [:environment] do
+    if FeatureSet.branston_retention_scheduling.enabled?
+      puts "\n*** Adding Retention Schedules to existing closed Branston cases ***"
 
-    def case_with_retention_schedule(case_type:, state:, planned_erasure_date:)
-      kase = FactoryBot::create(
-        case_type, 
-        retention_schedule: 
-        RetentionSchedule.new( 
-          state: state, 
-          planned_erasure_date: planned_erasure_date 
-        ) 
-      )
-      kase.save
-      kase
-    end
+      errors = []
+      off_sar_count = 0
+      off_sar_complaint_count = 0
+      already_had_rs = 0
+
+      percent = (Case::SAR::Offender.where(current_state: :closed).count / 100).round
+      percent_counter = 0
+      progressbar = ProgressBar.create
+
+      
+      Case::SAR::Offender.where(current_state: :closed).find_in_batches do | kase_group |
```

Comment:
If you want to save an inner loop, you can use `[#find_each](https://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each)` which uses find_in_batches behind the scenes (fault batch size of 1000).